File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task73_GEMINI_gemini-1.5-pro-001.java
import java.util.Scanner;

public class CheckPassword {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter password: ");
        String password = scanner.nextLine();
        if (isValidAdminPassword(password)) {
            System.out.println("Valid password");
        } else {
            System.out.println("Invalid password");
        }
    }

    public static boolean isValidAdminPassword(String password) {
        if (password.length() < 10) {
            return false;
        }
        if (!password.matches(".*[A-Z].*")) {
            return false;
        }
        if (!password.matches(".*[a-z].*")) {
            return false;
        }
        if (!password.matches(".*\\d.*")) {
            return false;
        }
        if (!password.matches(".*[!@#$%^&*()].*")) {
            return false;
        }
        return true;
    }
}

File: dataset/output/GEMINI_gemini-1.5-pro-001/Task73_GEMINI_gemini-1.5-pro-001.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int isValidAdminPassword(char *password);

int main() {
    char password[100];
    printf("Enter password: ");
    fgets(password, sizeof(password), stdin);
    password[strcspn(password, "
")] = 0; // remove trailing newline

    if (isValidAdminPassword(password)) {
        printf("Valid password
");
    } else {
        printf("Invalid password
");
    }
    return 0;
}

int isValidAdminPassword(char *password) {
    int hasUpper = 0, hasLower = 0, hasDigit = 0, hasSpecial = 0;
    if (strlen(password) < 10) {
        return 0;
    }
    for (int i = 0; password[i] != '\0'; i++) {
        if (isupper(password[i])) {
            hasUpper = 1;
        } else if (islower(password[i])) {
            hasLower = 1;
        } else if (isdigit(password[i])) {
            hasDigit = 1;
        } else if (strchr("!@#$%^&*()", password[i]) != NULL) {
            hasSpecial = 1;
        }
    }
    return hasUpper && hasLower && hasDigit && hasSpecial;
}